superset/templates/superset/basic.html

```diff
@@ -1,124 +1,139 @@
-              data-dismiss="modal"
-              aria-label="Close"
-              data-test="modal-header-close-button"
-            >
-              <span aria-hidden="true">&times;</span>
-            </button>
-            <h4 data-test="modal-title" class="modal-title"></h4>
-          </div>
-          <div data-test="modal-body" class="modal-body"></div>
-          <div data-test="modal-footer" class="modal-footer">
-            <button type="button" class="btn btn-default" data-dismiss="modal">
-              Close
-            </button>
+    <body {% if standalone_mode %}class="standalone" {% endif %}>
+      {% block navbar %} {% if not standalone_mode %} {% include
+      'appbuilder/navbar.html' %} {% endif %} {% endblock %} {% block body %}
+      <div id="app" data-bootstrap="{{ bootstrap_data }}">
+        <img
+          src="{{ assets_prefix }}/static/assets/images/loading.gif"
+          style="
+            width: 50px;
+            position: absolute;
+            top: 50%;
+            left: 50%;
+            transform: translate(-50%, -50%);
+          "
+        />
+      </div>
+      {% endblock %}
+
+      <!-- Modal for misc messages / alerts  -->
+      <div
+        class="misc-modal modal fade"
+        tabindex="-1"
+        role="dialog"
+        aria-labelledby="myModalLabel"
+      >
+        <div class="modal-dialog" role="document">
+          <div class="modal-content" data-test="modal-content">
+            <div class="modal-header" data-test="modal-header">
+              <button
+                type="button"
+                class="close"
+                data-dismiss="modal"
+                aria-label="Close"
+                data-test="modal-header-close-button"
+              >
+                <span aria-hidden="true">&times;</span>
+              </button>
+              <h4 data-test="modal-title" class="modal-title"></h4>
+            </div>
+            <div data-test="modal-body" class="modal-body"></div>
+            <div data-test="modal-footer" class="modal-footer">
+              <button type="button" class="btn btn-default" data-dismiss="modal">
+                Close
+              </button>
+            </div>
           </div>
         </div>
       </div>
-    </div>
 
-    {% block tail_js %} {% if not standalone_mode %} {{ js_bundle('menu') }} {%
-    endif %} {% if entry %} {{ js_bundle(entry) }} {% endif %} {% include
-    "tail_js_custom_extra.html" %} {% endblock %}
-  </body>
-</html>
+      {% block tail_js %} {% if not standalone_mode %} {{ js_bundle('menu') }} {%
+      endif %} {% if entry %} {{ js_bundle(entry) }} {% endif %} {% include
+      "tail_js_custom_extra.html" %} {% endblock %}
+
+      <!-- Handle deferred stylesheets -->
+      <script>
+        document.addEventListener('DOMContentLoaded', function() {
+          var preloadLinks = document.querySelectorAll('link[rel="preload"][as="style"]');
+
+          preloadLinks.forEach(function(link) {
```

Comment:
load event is triggered for the whole page, so we don't need a separate event listener for each link

Also, I wonder if we can just swap `rel` on DOMContentLoaded event